pkg/reconciler/v1alpha1/image/image.go

```diff
@@ -125,12 +125,20 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 
 func (c *Reconciler) reconcileImage(image *v1alpha1.Image) (*v1alpha1.Image, error) {
 	builder, err := c.getBuilder(image)
-	if k8serrors.IsNotFound(err) {
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return nil, err
+	} else if k8serrors.IsNotFound(err) {
 		image.Status.Conditions = image.BuilderNotFound()
 		image.Status.ObservedGeneration = image.Generation
 		return image, nil
 	}
 
+	if !builder.Ready() {
```

Comment:
Doing this check here will short circuit all the the other image reconciliation logic. 
While a builder 'is not ready': 
- this will cause existing Conditions to be overwritten 
- prevent valuable updates from taking place to the existing image object (like latestImage) 

Could we append this to the conditions when processing the build? 
Perhaps append on this line?https://github.com/pivotal/kpack/blob/d2b6381a892a9a1659ea2302d276f9167ffff918/pkg/apis/build/v1alpha1/image_reconcile_build.go#L56

This would also enable use to add a condition when the sourceresolver is not ready.